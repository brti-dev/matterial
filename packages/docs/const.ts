import packageJson from './package.json'
import { Color } from 'interfaces/theme'

export const COLORS: Color[] = [
  'primary',
  'secondary',
  'error',
  'warning',
  'info',
  'success',
  'dark',
  'light',
  'contrast',
]

export const TITLE = 'Matterial UI'

export const DESCRIPTION = packageJson.description

export const PACKAGE = 'matterial'
