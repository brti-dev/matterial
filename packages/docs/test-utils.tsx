import React, { FC, ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

function ThemeProvider({ theme, children }) {
  return <>{children}</>
}

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme="light">{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
