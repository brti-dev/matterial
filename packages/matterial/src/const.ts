import { Color, Accent } from './interfaces/theme'

export const COLORS: Color[] = [
  'primary',
  'secondary',
  'error',
  'warning',
  'info',
  'success',
  'dark',
  'light',
]

export const ACCENTS: Accent[] = [
  'accent-1',
  'accent-2',
  'accent-3',
  'accent-4',
  'accent-5',
  'accent-6',
  'accent-7',
  'accent-8',
]
