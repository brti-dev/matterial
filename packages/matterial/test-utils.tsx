import { FC, ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

// Example of wrapping a test
function ThemeProvider({ theme, children }: any) {
  return <div className={`theme--${theme}`}>{children}</div>
}

const AllTheProviders: FC = ({ children }: any) => {
  return <ThemeProvider theme="light">{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
