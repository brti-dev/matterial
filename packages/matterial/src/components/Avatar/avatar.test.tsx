import React from 'react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

import { render } from '../../../../../test-utils'
import { COLORS } from '../../const'
import { Avatar, AvatarGroup } from '.'
import {
  OverloadedElement,
  OverloadedElementProps,
} from '../../interfaces/OverloadedElement'

describe('avatar', () => {
  test('should work with different colors', () => {
    const tree = renderer
      .create(
        <>
          {[...COLORS, 'hotpink'].map(color => (
            <Avatar alt={color} color={color} key={color}>
              {color}
            </Avatar>
          ))}
        </>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render initials and label properly', () => {
    const { getByText } = render(<Avatar alt="Barry Lyndon">BL</Avatar>)

    expect(getByText('BL')).toHaveAttribute('aria-label', 'Barry Lyndon')
  })

  test('should not label when similar text is present', () => {
    const { getByText } = render(<Avatar alt="BL">BL</Avatar>)

    expect(getByText('BL')).not.toHaveAttribute('aria-label')
  })

  test('should render an image', () => {
    const alt = 'Dan Abramov'
    const src = 'https://bit.ly/dan-abramov'
    const { getAllByRole } = render(<Avatar alt={alt} src={src} />)

    const img = getAllByRole('img')
    expect(img).toHaveLength(2)
    expect(img[1]).toHaveAttribute('alt', alt)
    expect(img[1]).toHaveAttribute('src', src)
  })

  test('should size correctly by default and when given a `size` value', () => {
    const { getByText } = render(
      <AvatarGroup>
        <Avatar alt="a">a</Avatar>
        <Avatar alt="b" size={20}>
          b
        </Avatar>
      </AvatarGroup>
    )

    expect(getByText('a')).toHaveStyle({ '--size': '40px' })
    expect(getByText('b')).toHaveStyle({ '--size': '20px' })
  })

  test('should overload element root when given `as` prop', () => {
    const Foo = ({ foo, ...props }: { foo: string }) => (
      <div data-foo={foo} {...props} />
    )

    const { getByText } = render(
      <Avatar alt="foo" as={Foo} foo="foo">
        BL
      </Avatar>
    )

    expect(getByText('BL')).toHaveAttribute('data-foo', 'foo')
  })

  test('should function as overloaded element', () => {
    type User = {
      username: string
      id: number
      image: `cloudinary:${string}`
    }
    type Props = {
      user: User
      children?: React.ReactNode
    } & OverloadedElementProps

    const user: User = {
      username: 'User987',
      id: 123,
      image: 'cloudinary:foo.jpg',
    }

    const ProfileImage: OverloadedElement<Props> = ({
      user,
      as: Component = 'div',
      ...props
    }: Props) => <Component src={'foo.jpg'} alt={user.username} {...props} />

    const { getByLabelText } = render(
      <ProfileImage as={Avatar} user={user}>
        BL
      </ProfileImage>
    )

    expect(getByLabelText(user.username)).toHaveAttribute('role', 'img')
  })
})

describe('avatar group', () => {
  test('should render group', () => {
    const { getAllByRole } = render(
      <AvatarGroup>
        <Avatar alt="foo">F</Avatar>
        <Avatar alt="bar">B</Avatar>
      </AvatarGroup>
    )

    expect(getAllByRole('img')).toHaveLength(2)
  })

  test('should hide excess avatars', () => {
    const { getAllByRole } = render(
      <AvatarGroup max={2}>
        <Avatar alt="foo">F</Avatar>
        <Avatar alt="bar">B</Avatar>
        <Avatar alt="baz">Z</Avatar>
        <Avatar alt="fiz">Z</Avatar>
      </AvatarGroup>
    )

    expect(getAllByRole('img')).toHaveLength(3)
  })

  test('should indicate total', () => {
    const { getByText } = render(
      <AvatarGroup total={100}>
        <Avatar alt="foo">F</Avatar>
        <Avatar alt="bar">B</Avatar>
      </AvatarGroup>
    )

    expect(getByText('+98')).toBeInTheDocument()
  })
})
