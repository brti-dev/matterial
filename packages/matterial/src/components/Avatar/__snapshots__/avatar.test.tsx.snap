// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar should work with different colors 1`] = `
Array [
  <div
    className="avatar variant--contained color--primary no-hover"
    role="img"
    style={
      Object {
        "--color": "var(--color-primary)",
        "--size": "40px",
      }
    }
  >
    primary
  </div>,
  <div
    className="avatar variant--contained color--secondary no-hover"
    role="img"
    style={
      Object {
        "--color": "var(--color-secondary)",
        "--size": "40px",
      }
    }
  >
    secondary
  </div>,
  <div
    className="avatar variant--contained color--dark no-hover"
    role="img"
    style={
      Object {
        "--color": "var(--color-dark)",
        "--size": "40px",
      }
    }
  >
    dark
  </div>,
  <div
    className="avatar variant--contained color--light no-hover"
    role="img"
    style={
      Object {
        "--color": "var(--color-light)",
        "--size": "40px",
      }
    }
  >
    light
  </div>,
  <div
    className="avatar variant--contained color--contrast no-hover"
    role="img"
    style={
      Object {
        "--color": "var(--color-contrast)",
        "--size": "40px",
      }
    }
  >
    contrast
  </div>,
  <div
    className="avatar variant--contained color--error no-hover"
    role="img"
    style={
      Object {
        "--color": "var(--color-error)",
        "--size": "40px",
      }
    }
  >
    error
  </div>,
  <div
    className="avatar variant--contained color--warning no-hover"
    role="img"
    style={
      Object {
        "--color": "var(--color-warning)",
        "--size": "40px",
      }
    }
  >
    warning
  </div>,
  <div
    className="avatar variant--contained color--info no-hover"
    role="img"
    style={
      Object {
        "--color": "var(--color-info)",
        "--size": "40px",
      }
    }
  >
    info
  </div>,
  <div
    className="avatar variant--contained color--success no-hover"
    role="img"
    style={
      Object {
        "--color": "var(--color-success)",
        "--size": "40px",
      }
    }
  >
    success
  </div>,
  <div
    className="avatar variant--contained color--hotpink no-hover"
    role="img"
    style={
      Object {
        "--color": "hotpink",
        "--size": "40px",
      }
    }
  >
    hotpink
  </div>,
]
`;
